spec/services/tasks/dimensions_reporting_service_spec.rb

```diff
@@ -0,0 +1,128 @@
+    allow(AdminSet).to receive(:where).with(title: 'Open_Access_Articles_and_Book_Chapters').and_return([admin_set])
+    stub_request(:head, 'https://test-url.com/')
+      .to_return(status: 200, headers: { 'Content-Type' => 'application/pdf' })
+    stub_request(:get, 'https://test-url.com/')
+      .to_return(body: pdf_content, status: 200, headers: { 'Content-Type' => 'application/pdf' })
+    allow(ingest_service).to receive(:process_publication).and_call_original
+    allow(ingest_service).to receive(:process_publication).with(satisfy { |pub| failing_publication_sample[:publications].include?(pub) }).and_raise(StandardError, test_err_msg)
+    marked_for_review_sample
+    # stub virus checking
+    allow(Hyrax::VirusCheckerService).to receive(:file_has_virus?) { false }
+    # stub longleaf job
+    allow(RegisterToLongleafJob).to receive(:perform_later).and_return(nil)
+    # stub FITS characterization
+    allow(CharacterizeJob).to receive(:perform_later)
+    ingested_publications
+  end
+
+  describe '#generate_report' do
+    it 'generates a report for ingest dimensions publications' do
+      report = service.generate_report
+      headers = report[:headers]
+      expect(report[:subject]).to eq('Dimensions Ingest Report for May 21, 2024 at 10:00 AM UTC')
+      expect(headers[:reporting_message]).to eq('Reporting publications from dimensions ingest at May 21, 2024 at 10:00 AM UTC by admin.')
+      expect(headers[:admin_set]).to eq('Admin Set: Open_Access_Articles_and_Book_Chapters')
+      expect(headers[:total_publications]).to eq("Total Publications: #{test_publications.length}")
+      expect(headers[:successfully_ingested]).to eq("\nSuccessfully Ingested: (#{successful_publication_sample[:publications].length} Publications)")
+      expect(headers[:marked_for_review]).to eq("\nMarked for Review: (#{marked_for_review_sample[:publications].length} Publications)")
+      expect(headers[:failed_to_ingest]).to eq("\nFailed to Ingest: (#{failing_publication_sample[:publications].length} Publications)")
+    end
+  end
+
+  describe '#extract_publication_info' do
+    def expect_publication_info(info_array, sample_array, failed, sample_start_index)
+      info_array.each_with_index do |info, i|
+
+        expect(info[:title]).to eq(sample_array[i]['title'])
+        expect(info[:id]).to eq(sample_array[i]['id'])
+        expect(info[:url]).to eq("#{ENV['HYRAX_HOST']}/concern/articles/#{sample_array[i]['article_id']}?locale=en") if !failed
+        expect(info[:error]).to eq("StandardError - #{test_err_msg}") if failed
+        if failed
+          expect(info[:pdf_attached]).to be_nil
+        else
+          # Offsetting the index if the sample start index is odd
+          offset_index = sample_start_index.even? ? i : i - 1
+          expect(info[:pdf_attached]).to eq('No') if offset_index.even?
+          expect(info[:pdf_attached]).to eq('Yes') if offset_index.odd?
+        end
```

Comment:
I think it'd be clearer to have a single conditional versus a conditional plus inline conditionals
```suggestion
        if failed
          expect(info[:error]).to eq("StandardError - #{test_err_msg}")
          expect(info[:pdf_attached]).to be_nil
        else
          expect(info[:url]).to eq("#{ENV['HYRAX_HOST']}/concern/articles/#{sample_array[i]['article_id']}?locale=en")
          # Offsetting the index if the sample start index is odd
          offset_index = sample_start_index.even? ? i : i - 1
          expect(info[:pdf_attached]).to eq('No') if offset_index.even?
          expect(info[:pdf_attached]).to eq('Yes') if offset_index.odd?
        end
```